tests/database_test_utils.py

```diff
@@ -1,113 +1,28 @@
-        if specification["table"] == "task":
-            default_specification["title"] = "Test task " + str(specification["id"])
-            default_specification["description"] = "Test task description " + str(specification["id"])
-            default_specification["sort_order"] = specification["id"]
-            default_specification["status_id"] = 1
-        elif specification["table"] == "status":
-            default_specification["name"] = "Status " + str(specification["id"])
-            default_specification["sort_order"] = specification["id"]
-        return default_specification | specification
-
-    def _fix_missing_relations(self, specification: dict[str, Any]) -> None:
-        """
-        If a dictionary with a known table is given, like a task, the related item,
-        for example a status, will be created in the database if the id does not
-        exist yet. We do this to comply with database relationships and make it
-        easier to set up tests with minimal effort.
-        :param specification: a dictionary with a table specification
-        :return: None
-        """
-        if specification["table"] == "task":
-            status_specification = {"id": specification["status_id"], "table": "status"}
-            if not self.item_exists(status_specification):
-                self.init([status_specification])
-
-    def get_items(self, specification: dict[str, str | int]) -> Any:
-        """
-        Create a query based on the dictionary specification and return the result
-        :param specification: a dictionary with a table specification
-        :return: the results of the query
-        """
-        values = ", ".join(
-            key + "=" + str(val) if str(val).isnumeric() else str('"' + val + '"')  # type: ignore
-            for key, val in specification.items()  # type: ignore
-            if key != "table"  # type: ignore
-        )
-        table = specification["table"]
-        statement = f"SELECT * FROM {table} WHERE {values}"  # noqa S608
-        return self.session.exec(text(statement)).all()  # type: ignore
-
-    def item_exists(self, specification: dict[str, Any]) -> bool:
-        """
-        Check if an item exists in the database with the table and id given
-        in the dictionary
-        :param specification: a dictionary with a table specification
-        :return: True if the item exists in the database, False otherwise
-        """
-        result = self.get_items(specification)
-        return len(result) != 0
-
-    def init(self, specifications: list[dict[str, str | int]] | None = None) -> None:
-        """
-        Given an array of specifications, create the database entries.
-
-        Example: [{'table': 'task', 'id': 1 'title': 'Test task 1', 'description': 'Test task description 1'}]
+    def __del__(self):
+        for model in self.models:
+            self.session.delete(model)
+        self.session.commit()
```

Comment:
As mentioned above, the __del__ function gives strange 'detached' errors on objects, possibly because it is called outside the 'scope' (by garbage collection). Suggestion to make it an explicit call. Also on some occassions it seems the models get deleted twice; to avoid that, using a while / models.pop solves that problem.
```suggestion
    def reset(self):
        while self.models:
            self.session.delete( self.models.pop())
        self.session.commit()
```

Reply:
Ah ok.